riff-raff/app/controllers/DeployController.scala

```diff
@@ -281,4 +285,31 @@ class DeployController(deployments: Deployments, prismLookup: PrismLookup, deplo
     }
   }
 
+  def deployFiles(projectName: String, id: String) = AuthAction { implicit request =>
+    def pathAndContent(artifact: S3Artifact): Option[(S3Artifact, S3Path, String)] = {
```

Comment:
Ah, I see it is impure :-/

This is ok so these are minor points, but this controller method feels alarming to me. It looks like the main goal of this first section is to try both deploy object paths, but that's hidden away in a function that transforms a few bits of data into a format you use.
Strange as well, that they get transformed into a special format only to be immediately matched on. Would it not be better to just have these be in scope? A for-comp seems like it might be a good fit for doing this.

Related, `deployObject` seems like a redundant name and I suspect it only exists to try and help explain why we even have this (potentially unnecessary) intermediate structure.

It feels like with a bit of thought this could be significantly simpler.